src/libraries/System.Private.CoreLib/src/System/IO/Path.Windows.cs

```diff
@@ -208,25 +208,57 @@ static uint GetTempPathW(int bufferLen, ref char buffer)
         // name on disk.
         public static string GetTempFileName()
         {
-            var tempPathBuilder = new ValueStringBuilder(stackalloc char[PathInternal.MaxShortPath]);
-
-            GetTempPath(ref tempPathBuilder);
-
-            var builder = new ValueStringBuilder(stackalloc char[PathInternal.MaxShortPath]);
-
-            uint result = Interop.Kernel32.GetTempFileNameW(
-                ref tempPathBuilder.GetPinnableReference(), "tmp", 0, ref builder.GetPinnableReference());
-
-            tempPathBuilder.Dispose();
-
-            if (result == 0)
-                throw Win32Marshal.GetExceptionForLastWin32Error();
-
-            builder.Length = builder.RawChars.IndexOf('\0');
+            // Avoid GetTempFileNameW because it is limited to 0xFFFF possibilities, which both
+            // means that it may have to make many attempts to create the file before
+            // finding an unused name, and also that if an app "leaks" such temp files,
+            // it can prevent GetTempFileNameW succeeding at all.
+            //
+            // To make this a little more robust, generate our own name with more
+            // entropy. We could use GetRandomFileName() here, but for consistency
+            // with Unix and to retain the ".tmp" extension we will use the "tmpXXXXXX.tmp" pattern.
+            // Using 32 characters for convenience, that gives us 32^^6 ~= 10^^9 possibilities,
```

Comment:
Nit (or was this by intention?):
```suggestion
            // Using 32 characters for convenience, that gives us 32^6 ~= 10^9 possibilities,
```

Reply:
Intentionally as in C# ^ is XOR. I have seen ^^ sometimes used for exponentiation.